.circleci/config.yml

```diff
@@ -326,4 +326,9 @@ commands:
             if git log -1 "$CIRCLE_SHA1" | grep -q '\[do not deploy\]'; then
                 echo "Skipping remaining steps in this job: deployment was disabled for this commit."
                 circleci-agent step halt
+
+                # No need to deploy, just cancel the rest of jobs of the workflow
```

Comment:
Do you mind adding a link to documentation for this API endpoint? Does it cancel all jobs in the workflow right away or does it wait for running jobs to complete?

Reply:
Sure, will do.

I couldn't find details in the API doc about how the cancellation is done. In this case, both `docker-image-publish-stage` and `docker-image-publish-prod` requires other jobs to be run first, and we cancel the workflow within those two jobs, so I think there should be no other jobs running at that point.

In fact, the `skip-if-do-not-deploy` command should cancel the workflow when `docker-image-publish-stage` gets run.